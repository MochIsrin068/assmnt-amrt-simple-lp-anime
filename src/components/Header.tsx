import { TPropsHeader } from "../types/components.types";
import IconBack from "../assets/arrow-back.png";

export default function Header({
  isHeaderWithAction,
  backgroundImageUrl,
  onAction,
}: TPropsHeader) {
  return (
    <>
      <header
        data-testid="header-container"
        className={`header ${
          isHeaderWithAction ? "--header-action" : "--header-home"
        }`}
        style={{
          backgroundImage: isHeaderWithAction
            ? `url(${backgroundImageUrl})`
            : "",
        }}
      >
        {isHeaderWithAction ? (
          <div className="header__back-menu">
            <span onClick={onAction} data-testid="header-back-menu">
              <img src={IconBack} alt="<" data-testid="icon-arrow-back" />
              Back
            </span>
          </div>
        ) : (
          <div className="header__jumbotron">
            <h1>The Anime Movie</h1>
            <p>
              The Anime Movie is a web-based application to see what movies are
              trending and popular that you can use as a reference to watch with
              your special someone :)
            </p>
          </div>
        )}
      </header>
      ;
    </>
  );
}
