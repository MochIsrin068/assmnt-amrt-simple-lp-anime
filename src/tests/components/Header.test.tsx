/* eslint-disable testing-library/await-async-utils */
import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import Header from "../../components/Header";

describe("Perform testing component Header", () => {
  // Handle error : TypeError: window.matchMedia is not a function
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      value: jest.fn(() => ({
        matches: true,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      })),
    });
  });

  it("Perform snapshots test Header", () => {
    const view = render(<Header isHeaderWithAction={false} />);
    expect(view).toMatchSnapshot();
  });

  it("test background image header if show on detail page", () => {
    const onActionSpy = jest.fn();
    render(
      <Header
        isHeaderWithAction={true}
        backgroundImageUrl="banner-detail.png"
        onAction={onActionSpy}
      />
    );
    const container = screen.getByTestId("header-container");
    expect(container).toHaveStyle(`background: url(banner-detail.png)`);
  });

  it("test class attribute header if show on home page", () => {
    render(<Header isHeaderWithAction={false} />);
    const container = screen.getByTestId("header-container");
    expect(container).toHaveAttribute("class", `header --header-home`);
  });

  it("test class attribute header if show on detail page", () => {
    const onActionSpy = jest.fn();
    render(
      <Header
        isHeaderWithAction={true}
        backgroundImageUrl="banner-detail.png"
        onAction={onActionSpy}
      />
    );
    const container = screen.getByTestId("header-container");
    expect(container).toHaveAttribute("class", `header --header-action`);
  });

  it("test button back or text showed header if show on home page", () => {
    render(<Header isHeaderWithAction={false} />);
    const container = screen.getByTestId("header-container");
    expect(container).toHaveAttribute("class", `header --header-home`);
  });

  it("test button back func header if show on detail page", () => {
    const onActionSpy = jest.fn();
    render(
      <Header
        isHeaderWithAction={true}
        backgroundImageUrl="banner-detail.png"
        onAction={onActionSpy}
      />
    );
    fireEvent.click(screen.getByTestId("header-back-menu"));
    expect(onActionSpy).toHaveBeenCalled();
  });

  it("test button back image / icon header if show on detail page", () => {
    const onActionSpy = jest.fn();
    render(
      <Header
        isHeaderWithAction={true}
        backgroundImageUrl="banner-detail.png"
        onAction={onActionSpy}
      />
    );
    const container = screen.getByTestId("icon-arrow-back");
    expect(container).toHaveAttribute("src", `arrow-back.png`);
  });

  it("test label button back header if show on home page", () => {
    const onActionSpy = jest.fn();
    render(
      <Header
        isHeaderWithAction={true}
        backgroundImageUrl="banner-detail.png"
        onAction={onActionSpy}
      />
    );
    waitFor(() => expect(screen.getByText(/Back/i)).toBeInTheDocument());
  });

  it("test title and subtitle header if show on home page", () => {
    render(<Header isHeaderWithAction={false} />);
    waitFor(() =>
      expect(screen.getByText(/The Anime Movie/i)).toBeInTheDocument()
    );

    waitFor(() =>
      expect(
        screen.getByText(
          "The Anime Movie is a web-based application to see what movies aretrending and popular that you can use as a reference to watch with your special someone :)"
        )
      ).toBeInTheDocument()
    );
  });
});
