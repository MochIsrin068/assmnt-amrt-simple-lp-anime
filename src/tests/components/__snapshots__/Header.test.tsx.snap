// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Perform testing component Header Perform snapshots test Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header --header-home"
        data-testid="header-container"
      >
        <div
          class="header__jumbotron"
        >
          <h1>
            The Anime Movie
          </h1>
          <p>
            The Anime Movie is a web-based application to see what movies are trending and popular that you can use as a reference to watch with your special someone :)
          </p>
        </div>
      </header>
      ;
    </div>
  </body>,
  "container": <div>
    <header
      class="header --header-home"
      data-testid="header-container"
    >
      <div
        class="header__jumbotron"
      >
        <h1>
          The Anime Movie
        </h1>
        <p>
          The Anime Movie is a web-based application to see what movies are trending and popular that you can use as a reference to watch with your special someone :)
        </p>
      </div>
    </header>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
